Simplenote/src/main/java/com/automattic/simplenote/TagsActivity.kt

```diff
@@ -0,0 +1,246 @@
+package com.automattic.simplenote
+
+import android.app.ActivityOptions
+import android.content.DialogInterface
+import android.content.Intent
+import android.os.Bundle
+import android.text.SpannableString
+import android.view.*
+import android.widget.*
+import androidx.activity.viewModels
+import androidx.annotation.DrawableRes
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.view.ContextThemeWrapper
+import androidx.appcompat.widget.SearchView
+import androidx.appcompat.widget.Toolbar
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.automattic.simplenote.databinding.ActivityTagsBinding
+import com.automattic.simplenote.utils.*
+import com.automattic.simplenote.viewmodels.TagsEvent
+import com.automattic.simplenote.viewmodels.TagsEvent.*
+import com.automattic.simplenote.viewmodels.TagsViewModel
+import com.automattic.simplenote.widgets.MorphSetup
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class TagsActivity : ThemedAppCompatActivity() {
+    private val viewModel: TagsViewModel by viewModels()
+
+    private var _binding: ActivityTagsBinding? = null
+    private val binding get() = _binding!!
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        _binding = ActivityTagsBinding.inflate(layoutInflater)
```

Comment:
Warning (⚠️): For activities you don't need to do the binding backing field trick so that to then make the binding null during `onDestroy()`. This is only required on fragment and in case you need that field since you need access to it from outside the `onCreate(...)`, just like you did for `TagDialogFragment`.

Actually, for `AddTagActivity` you did exactly what I am suggesting, so I am not sure why you didn't do the same for this activity as well.

As such, this can become `val binding: ActivityTagsBinding = ActivityTagsBinding.inflate(layoutInflater)` and the binding fields can be removed. Doing so will propagate further refactoring down the line, mainly making some more functions an extension function of the binding.

Reply:
Good point! Fixed in d3c92932c209f99075c7bc3ead2cbfcec4a0bcf6.

Reply:
Amazing, many thanks Danilo! 🕺 💃